lib/matplotlib/backend_bases.py

```diff
@@ -2149,6 +2150,15 @@ def print_figure(self, filename, dpi=None, facecolor=None, edgecolor=None,
 
         """
         self._is_saving = True
+        # Remove the figure manager, if any, to avoid resizing the GUI widget.
```

Comment:
Maybe `# Remove the figure manager, currently set in some backeneds (i.e. nbagg)...`.  I was looking at this and not understanding when `self.manager` ever got set.  Similarly above when you instantiate it, maybe a comment on what a manager is and where one is set?

Reply:
There is a FigureManagerBase class defined in the same file, and if you look at its init you'll see it creates the manager attribute on the canvas.  I think it's better to have a single doc that describes the interaction between all classes in backend_bases rather than document this in a disjoint, one-comment-at-a-time way... (but it is also true that that single doc doesn't exist yet)

Reply:
I'm still confused what backend I need to use to test this works, and how.  I assume I need to trigger a manual save.  This works fine on Qt5Agg, but did not check if manager is set in Qt5Agg.

Reply:
Set the savefig.dpi rcparam to some large number (say, three times your figure.dpi, just to be safe with hidpi stuff...) and then interactively trigger a save, while the window has a relatively small size on your desktop.  Depending on yourdesktop environment (specifically, how it handles fast window resize events), you may see the window flicker in size, first to, well, three times its current size, and then back.
This resize is handled by the canvas manager.  This PR disconnects the manager for the duration of the save, thus suppressing the flicker.

Reply:
Hmmm, I guess my machine is too fast - I don't get a flicker w/ Qt5Agg or any apparent resizing.

Reply:
Can you try on linux?  It may well be that OSX just detects and suppresses fast resizes.
May also help if you make the plot relatively slow to draw (e.g., `N=100000; scatter(rand(N), rand(N), s=rand(N))`), which may make the effect more apparent.

Reply:
No, I don't have a linux machine (set up for this).   I guess OSX is just clever - making the plot more ornate doesn't change anything.

Reply:
I assume @tacaswell checked that this works for him too?